File: kinotic-domain/src/main/java/org/kinotic/domain/api/model/log/LogQuery.java
package org.kinotic.domain.api.model.log;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

/**
 * Parameters for a historical log query: one workload's logs over a time range.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class LogQuery {

    /**
     * Id of the workload whose logs to return.
     */
    private String workloadId;

    /**
     * Start of the time range, epoch milliseconds (inclusive).
     */
    private long start;

    /**
     * End of the time range, epoch milliseconds (inclusive).
     */
    private long end;

    /**
     * Maximum number of log entries to return.
     */
    private int limit;
}

File: kinotic-js/workspace/packages/management-api/src/api/model/log/LogQuery.ts
/**
 * Parameters for a historical log query: one workload's logs over a time range.
 */
export interface LogQuery {

    /**
     * Organization whose workload's logs to return. Null names the platform's own, which only a
     * system participant may read.
     */
    organizationId: string | null

    /** Id of the workload whose logs to return. */
    workloadId: string

    /** Start of the time range, epoch milliseconds (inclusive). */
    start: number

    /** End of the time range, epoch milliseconds (inclusive). */
    end: number

    /** Maximum number of log entries to return. */
    limit: number
}
